Code/GraphMol/FileParsers/testGeneralFileReader.cpp

```diff
@@ -0,0 +1,223 @@
+#include <RDGeneral/test.h>
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MonomerInfo.h>
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <memory>
+#include <RDGeneral/FileParseException.h>
+#include <RDGeneral/RDLog.h>
+#include "GeneralFileReader.h"
+#include <boost/iostreams/device/file.hpp>
+#include <boost/iostreams/filtering_stream.hpp>
+
+namespace io = boost::iostreams;
+using namespace RDKit;
+
+
+void testSdf(){
+  // Open uncompressed SDF file format
+  std::string rdbase = getenv("RDBASE");
+  std::string fname = rdbase + "/Code/GraphMol/FileParsers/test_data/NCI_aids_few.sdf";
+  struct SupplierOption opt_sdf;
+  GeneralFileReader gfr(fname, opt_sdf);
+  MolSupplier * sdsup = gfr.getSupplier();
```

Comment:
because you own the memory for this supplier, you need to `delete` it at the end of the test.
This comment applies to all the tests in this file.